src/ZkFold/Base/Algebra/Polynomials/Multivariate/Polynomial.hs

```diff
@@ -38,15 +39,20 @@ instance Polynomial c i j => FromPolynomial c i j m [(c, M i j m)] where
     fromPolynomial = id
 
 class Polynomial c i j => ToPolynomial c i j m p where
-    toPolynomial   :: [(c, M i j m)] -> Maybe p
+    toPolynomial   :: [(c, M i j m)] -> p
```

Comment:
The motivation behind `Maybe` was that this operation is fallible e.g. because accumulated powers become too large. Do we throw an exception here now instead?

Reply:
@TurtlePU There is only one instance of `ToPolynomial`, we just add `Just` there. No I think that it's better to remove them, they add an unnecessary level of complexity.

Reply:
If there's only 1 instance, why bother with a typeclass?